### ISSUE TITLE
I can not install any code editor in ubuntu 22.04 that in umm !!

### ISSUE THREAD

**Issue by SKbarbon** [utmapp/UTM#4313]
I have this errors:

>  sudo apt install code
>> E: Unable to locate package code

And when I tried to use this another command, I got an error:
> sudo snap install code --classic
>> error: snap "code" is not availble on stable for this architecture (arm64) but exist on other architectures (arm64).

My device is MacBook Pro with M1 Pro chip.

and I give ubuntu:
- 8GB RAB.
- 65GB storage.
- 4 core -multicore force-.

is there any thing wrong ? please help me !

**Comment by mikeactual** [comment]
Yeah, it seems like linux graphics supprt for the ARM64 architecture is poor. It may be an upstream problem from UTM though. I asked for clarification on another thread a few days ago but haven't heard anything yet.

In any case your workaround is to install vs code on your mac with microsoft's remote development extension. From there you can make a remote conection to the ubuntu vm and code on it just as if you would were running vs code local on the vm. You don't really need ubuntu desktop installed on the vm at all in this scenario.

If you really just want to install vs code locally on the ubuntu vm you'll need to download the arm64 build from the vs code website and install per the instructions there. I've done that too. vs code installs but all I get is a black window when I execute it.

Good luck!

**Comment by goshdarnitnick** [comment]
I've found that configuring an un-accelerated emulated display card such as `virtio-gpu-pci` or `virtio-rambf` (choose the one without the GPU supported label) in the Display settings fixes the black screen issue on certain GUI apps (seems like electron is a common theme). I haven't looked into this much, but I'd assume certain calls to the GPU acceleration code just aren't working properly right now. Unfortunately this will reduce your GPU performance since it's now being slowly emulated.
